lib/aws_beanstalk_tunkki.rb

```diff
@@ -82,14 +82,15 @@ def set_aws_clients
   end
 
   def get_aws_keys
-    case @environment
-    when 'prod', 'st'
-      [ENV['AWS_ACCESS_KEY_ID_PROD'], ENV['AWS_SECRET_ACCESS_KEY_PROD']]
-    when 'dev', 'ft'
-      [ENV['AWS_ACCESS_KEY_ID_DEV'], ENV['AWS_SECRET_ACCESS_KEY_DEV']]
-    when 'sb'
-      [ENV['AWS_ACCESS_KEY_ID_SB'], ENV['AWS_SECRET_ACCESS_KEY_SB']]
-    end
+    [ENV['AWS_ACCESS_KEY_ID'], ENV['AWS_SECRET_ACCESS_KEY']]
+    #case @environment
+    #when 'prod', 'st'
+    #  [ENV['AWS_ACCESS_KEY_ID_PROD'], ENV['AWS_SECRET_ACCESS_KEY_PROD']]
+    #when 'dev', 'ft'
+    #  [ENV['AWS_ACCESS_KEY_ID_DEV'], ENV['AWS_SECRET_ACCESS_KEY_DEV']]
+    #when 'sb'
+    #  [ENV['AWS_ACCESS_KEY_ID_SB'], ENV['AWS_SECRET_ACCESS_KEY_SB']]
+    #end
```

Comment:
Mietin että tämä ei ole nyt taakseppäin yhteensopiva muutos, koska travisissa tämä ei toimi.
Toimisikohan tuo local deploy, jos poistaa rivin
> raise "AWS_PROFILE environment varible is not set!" if ENV['AWS_PROFILE'].nil?

niin tämä toimisi samaan aikaa travisissa ja GA:ssa?
Muistaakseni aws sdk käyttää oletuksena AWS_ACCESS_KEY_ID ja AWS_SECRET_ACCESS_KEY env muuttujia, jos muuta ei ole annettu.

Reply:
AWS SDK etsii avaimia ns. "Credential Provider Chain" avulla jossa se yrittää (_tässä kontekstissa relevantisti_) järjestyksessä:

1. SDK:lle suoraan määritetyt
2. Profiili resolvointi järjestyksessä: SDK suoraan määritetty -> AWS_PROFILE env -> AWS_DEFAULT_PROFILE env -> "default" string
3. Profiilin jälkeen se yrittää etsiä avaimia kyseisen profiilin alta ~/.aws/credentials tiedostosta (_johon GHA tapauksessa avaimet on kirjoitettu aws configure set määrityksillä_)

Eli toisin sanoen joo, tän pull requn voi mun mielestä yksinkertaistaa yhden rivin muutokseen jossa poistetaan tuo raise "AWS_PROFILE" checkkaus ja Github Actions deployt menee tuolla local asetuksella samalla tavalla kuin omalta koneelta ajettaessa.

Tuo AWS_PROFILE check on jäänne siltä ajalta kun siellä credu filussa saattoi olla useiden eri tilien avaimia eikä käytetty aina sitä "default":ia.